lib/g_searcher_web/plugs/set_current_user.ex

```diff
@@ -0,0 +1,25 @@
+defmodule GSearcherWeb.Plugs.SetCurrentUser do
+  @moduledoc """
+  Plug which sets the user to the connection if the user is authenticated
+  """
+
+  import Plug.Conn
+
+  alias GSearcher.Accounts
+
+  def init(_options), do: nil
+
+  def call(conn, _options) do
+    user_id = Plug.Conn.get_session(conn, :current_user_id)
+
+    if current_user = user_id && Accounts.get_by_user_id!(user_id) do
+      conn
+      |> assign(:current_user, current_user)
```

Comment:
`current_user` here is the ID value or the `current_user` object @bterone? I guess it's the `user_id`, should we change to use `current_user_id` to make that more descripite

Reply:
It does set the `current_user` here though, not the `current_user_id`. So we can fetch the user details later if we need to. 🙁

Reply:
Ah I see, the current_user = `user_id && Accounts.get_user_by_id(user_id)` it means